pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -29,7 +30,24 @@ func NewManager(client client.Client, nodeName string, policy nmstatev1alpha1.No
 	manager.logger = logf.Log.WithName("policy/conditions/manager").WithValues("enactment", nmstatev1alpha1.EnactmentKey(nodeName, policy.Name).Name)
 	return manager
 }
-
+func (m *Manager) NotifyNodeSelectorNotMatching(unmatchingLabels map[string]string) {
+	message := ""
+	if len(unmatchingLabels) > 0 {
+		message = fmt.Sprintf("Unmatching labels: %v", unmatchingLabels)
+	} else {
+		message = fmt.Sprintf("Cannot retrieve node %s", m.nodeName)
```

Comment:
This error is completely implicit. Meaning of zero unmatchingLabels should not be "cannot retrieve node".

Reply:
Let's refactor `MatchesThisNode` so it returns an error instead of bool, so if unmatching labels is empty but we  have an error we print that on the condition.

Reply:
Smart